vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc

```diff
@@ -0,0 +1,289 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <cmath>
+#include <utility>
+#include <vector>
+
+#include "base/strings/string_split.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/media/helper.h"
+#include "bat/ledger/internal/media/reddit.h"
+#include "net/http/http_status_code.h"
+#include "url/url_canon.h"
+#include "url/gurl.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace braveledger_media {
+
+MediaReddit::MediaReddit(bat_ledger::LedgerImpl* ledger): ledger_(ledger) {
+}
+
+MediaReddit::~MediaReddit() {
+}
+
+void MediaReddit::ProcessActivityFromUrl(
+    uint64_t window_id,
+    const ledger::VisitData& visit_data) {
+  if (visit_data.path.find("/user/") != std::string::npos) {
+    UserPath(window_id, visit_data);
+    return;
+  }
+  OnMediaActivityError(visit_data, window_id);
+}
+
+void MediaReddit::OnMediaActivityError(
+    const ledger::VisitData& visit_data,
+    uint64_t window_id) {
+  std::string url = REDDIT_TLD;
+  std::string name = REDDIT_MEDIA_TYPE;
+
+  if (!url.empty()) {
```

Comment:
nit: switch if and return, so that we don't have this big if

Reply:
Done